core/src/main/java/hudson/model/ParametersAction.java

```diff
@@ -326,7 +326,7 @@ public void onLoad(Run<?, ?> r) {
         for (ParameterValue v : this.parameters) {
             if (this.parameterDefinitionNames.contains(v.getName()) || isSafeParameter(v.getName())) {
                 filteredParameters.add(v);
-            } else if ("false".equalsIgnoreCase(shouldKeepFlag)) {
+            } else if (!"false".equalsIgnoreCase(shouldKeepFlag)) {
```

Comment:
Check the original motivation in the linked PR; the idea is to be effectively tri-state: Explicit `true`, explicit `false`, and undefined.